addons/website_livechat/static/src/legacy/website_livechat_chatbot_test_script.js

```diff
@@ -84,6 +84,7 @@ publicWidget.registry.livechatChatbotTestScript = publicWidget.Widget.extend({
         utils.set_cookie('im_livechat_previous_operator_pid', '', -1);
 
         return this._super(...arguments).then(() => {
+            owl.Component.env.messaging.update({ isInPublicLivechat: true });
```

Comment:
this.env possible?
Why is it not using the messaging value provider service? In which context is this "test script" ran?

Reply:
`PublicWidget` that are globally registered are started in a completely different way than services. As I said in https://github.com/odoo/odoo/pull/92786#discussion_r894636979, I can do some things to have `this.env` available for `LivechatButton`.